lib/state/connection/connection_cubit.dart

```diff
@@ -73,62 +63,73 @@ class ConnectionCubit extends Cubit<ConnectionState> {
 
   /// Connects to the web socket.
   ///
-  /// This function MUST NOT be called outside of this package.
+  /// This function `MUST NOT` be called outside of this package.
   Future<void> connect({
     ConnectionInformation? connectionInformation,
     bool printResponse = false,
   }) async {
-    if (state is! ConnectionConnectingState) {
-      emit(ConnectionConnectingState());
+    if (state is ConnectionConnectingState) {
+      return;
     }
 
+    emit(const ConnectionConnectingState());
+
+    await _api!.disconnect().timeout(_callTimeOut);
+
     if (connectionInformation != null) {
       _connectionInformation = connectionInformation;
     }
 
-    await _api!.disconnect().timeout(_callTimeOut);
-
     await _api!.connect(
       _connectionInformation,
       printResponse: printResponse,
+      onOpen: (UniqueKey uniqueKey) {
+        if (_uniqueKey == uniqueKey && state is! ConnectionConnectedState) {
```

Comment:
```suggestion
        if (_uniqueKey == uniqueKey) {
```
Can we skip checking state here?

Reply:
if we remove this condition `onOpen` will be called in every response.